api/src/main/java/org/openmrs/module/dhisconnector/api/impl/DHISConnectorServiceImpl.java

```diff
@@ -1380,6 +1380,19 @@ public String transformToDHISPeriod(Calendar startDate, Calendar endDate, String
 			endDate.set(Calendar.DAY_OF_YEAR, startDate.getActualMaximum(Calendar.DAY_OF_YEAR));
 			if (lastRun == null || !sdf.format(lastRun).equals(sdf.format(endDate.getTime())))
 				period = new SimpleDateFormat("yyyy").format(startDate.getTime());
+		} else if (ReportingPeriodType.FinancialApril.name().equals(periodType)) {
+			sdf = new SimpleDateFormat("yyyy");
+			// Set the start date to 1st of April
```

Comment:
please use https://docs.oracle.com/javase/7/docs/api/java/util/Calendar.html#APRIL and other specific values than direct integer values.